Lib/test/test_print.py

```diff
@@ -155,6 +155,15 @@ def test_string_with_excessive_whitespace(self):
 
         self.assertIn('print("Hello World", end=" ")', str(context.exception))
 
+    def test_string_with_stream_redirection(self):
+        import sys
+        python2_print_str = 'print >> sys.stderr, "message"'
+        with self.assertRaises(TypeError) as context:
+            exec(python2_print_str)
+
+        self.assertIn('Did you mean "print(<message>, '
+                'file=<output_stream>)', str(context.exception))
+
```

Comment:
@CuriousLearner Correct for the first case. For the second case, you'll want to implement a dummy class that does something like this:

    class OverrideRRShift:
        def __rrshift__(self, lhs):
            return 42 # Force result independent of LHS

    self.assertEqual(print >> OverrideRRShift(), 42)

That will specifically ensure that the special casing doesn't trigger when the rshift operation produces a sensible answer.

It would also be good to test that `print >> 5` *does* produce the new exception message (that covers the case where the RHS implements `__rrshift__` but returns `NotImplemented`)